internal/auth/authorizations.go

```diff
@@ -25,6 +27,13 @@ type State struct {
 	Expires        time.Time
 }
 
+var authStateCache map[string]State
+var authStateCacheMutex sync.RWMutex
+
+func init() {
+	authStateCache = make(map[string]State)
+}
```

Comment:
This is a good start, but I don't think we can keep this "global", I think we should attach it to some of the other structs we're using for auth.

Similarly, this might be addressed as a side effect, I'm not sure `QueryAuthd` is the right spot to do the caching.  It feels like a responsibility of a higher level abstraction, or at least a wrapper for `QueryAuthd`?